tests/test_core/test_responses_module.py

```diff
@@ -17,8 +17,8 @@ def setUp(self) -> None:
         if settings.TEST_SERVER_MODE:
             raise SkipTest("No point in testing responses-decorator in ServerMode")
 
-    @mock_s3
     @responses.activate
+    @mock_s3
```

Comment:
Mypy showed a typing error when mock_s3 came first. I'm not sure what the cause was, and that might be something worth checking out

Reply:
For future reference, the MyPY error:
`Type of decorated function contains type "Any" ("Callable[[VarArg(Any), KwArg(Any)], Any]")`

I _think_ this is due to the fact that the `responses`-decorator actually has `Any` in their typing declarations - see for example here:

https://github.com/getsentry/responses/blob/4fa24b7b2342cd756c9b9533d8036132eb11c91b/responses/__init__.py#L990

This method tests a very specific scenario where Moto's decorator comes first  - hence the name, `test_moto_first`. (The next test has the exact same body, but with the decorators reversed - `test_moto_second`).

So to ensure we're not losing any code/scenario coverage, I think this should be reversed. Maybe we can just add a `type: ignore` to get past the fact that `responses` is to loose in their annotations.

Reply:
Ah ha! That's what I get for fixing the type error without really paying attention to what it was. I'll stare at it a little longer and put the ignore pragma on here if I can't figure out another solution.